cli: some cleanups to the new package `clisqlclient` (3/)

knz opened the pull request:
First 6 commits are from #67457 and prior.
Please ignore those (they will disappear once the prefix PR is merged)

Only the last 3 commits belong to this PR..

This is part of a family of PRs that aim to encapsulate CLI functionality into finer grained packages.

knz committed:
cli: move SQL client code to `clisqlclient`, new package `clierror`

This commit leaves the code in an intermediate stage, where the unit
tests pass but maybe the linter is unhappy. This will be cleaned up in
the next commit.

Release note: None

knz committed:
clisqlclient: rename some files

Release note: None

knz committed:
clisqlclient: move InitialSQLConnectionError to own file

Release note: None

knz committed:
cli, clisqlclient: only uses public APIs

Release note: None

knz committed:
cli: properly handle conn close errors

Release note: None

knz committed:
clisqlclient: split code in smaller files

Release note: None

knz committed:
clisqlclient: `conn_util` -> `conn`

Release note: None

knz committed:
clisqlclient: avoid depending on formatting code in conn.go

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/67458)
<!-- Reviewable:end -->

otan commented on pkg/cli/clisqlclient/txn_shim.go:
```diff
@@ -0,0 +1,46 @@
+	conn *sqlConn
+}
+
+var _ crdb.Tx = sqlTxnShim{}
```
i'm somewhat amused we import all of cockroach-go just to ensure it obeys the interface. maybe we should move this assertion to the test to optimise out this import.

knz replied:
It's not just that, we actually use `crdb.ExecuteInTx` to implement `(*sqlConn).ExecTxn`.

knz committed:
cli: new format 'rawhtml' to auto-gen the settings table

Previously, the cluster setting markdown table was generated with a
`gen` sub-command that was trying to be too clever with the HTML
generator. This was making it complicated to move the table formatting
code to a separate package.

This commit simplifies the code by introducing a new output format
'rawhtml' (raw, as in, it does not escape HTML special characters
inside table cells).

Since this format is only used internally to auto-generate this table,
the help texts leave it undocumented.

Release note: None

knz committed:
clisqlclient: document the table output formats

Release note: None

knz commented:
TFYR!

bors r=otan

knz commented:
the bors run is currently flaking on #64866. I hope it gets retried.
